src/iocore/net/UnixNetVConnection.cc

```diff
@@ -418,6 +418,12 @@ write_to_net_io(NetHandler *nh, UnixNetVConnection *vc, EThread *thread)
       if (vc->write.enabled) {
         nh->write_ready_list.in_or_enqueue(vc);
       }
+      // If this was driven by a zero length read, signal complete when
+      // the handshake is complete. Otherwise set up for continuing read
+      // operations.
+      if (s->vio.ntodo() <= 0) {
+        vc->readSignalDone(VC_EVENT_READ_COMPLETE, nh);
```

Comment:
`VC_EVENT_READ_COMPLETE`? Not `VC_EVENT_WRITE_COMPLETE`?

Reply:
Great catch! I copied from the read operation of the SSLNetVConnection, but this is for the `do_io_write`. Let me try out the patch with WRITE_COMPLETE and verify that addresses the failure as well. It almost certainly will since the ConnectingEntry handles both the read and write complete events in the same switch case block.